codecarbon/core/api_client.py

```diff
@@ -0,0 +1,100 @@
+"""
+
+Based on https://kernelpanic.io/the-modern-way-to-call-apis-in-python
+
+TODO : use async call to API
+"""
+# from httpx import AsyncClient
+import dataclasses
+import json
+import time
+from datetime import datetime
+
+import requests
+
+from codecarbon.core.schemas import EmissionCreate, RunCreate
+from codecarbon.external.logger import logger
+
+# from codecarbon.output import EmissionsData
+
+
+class ApiClient:  # (AsyncClient)
+    """
+    This class call the Code Carbon API
+    Note : The project, team and organization must have been created in the interface.
+    """
+
+    run_id = None
+    _previous_call = time.time()
+
+    def __init__(self, experiment_id, endpoint_url, api_key):
+        """
+        :project_id: ID of the existing project
+        :api_ley: Code Carbon API_KEY
+        """
+        # super().__init__(base_url=endpoint_url) # (AsyncClient)
+        self.url = endpoint_url
+        self.experiment_id = experiment_id
+        self.api_key = api_key
+        self._create_run(self.experiment_id)
+
+    def add_emission(self, carbon_emission: dict):
+        self._previous_call = time.time()
+        if self.run_id is None:
+            # TODO : raise an Exception ?
+            logger.error(
+                "add_emissionadd_emission need a run_id : the initial call may have failed. Retrying..."
+            )
+            self._create_run(self.experiment_id)
+        if carbon_emission["duration"] < 1:
+            logger.warning(
+                "Warning : emission not send because of a duration smaller than 1."
+            )
+            return False
+        emission = EmissionCreate(
+            timestamp=datetime.now().isoformat(),
```

Comment:
We might want to send UTC timestamps

Reply:
Agree, should we send local timezone identifier to let the front end app reconstruct a correct time ?
It may be stored at the run / run config level

Reply:
I have made a change to send: `2021-06-07T22:31:31.142202+02:00` but in the database it seems to be just in UTC : `2021-06-07T20:31:31.142202`
Is it OK for you ?